test/test_ldap_connection.rb

```diff
@@ -9,6 +9,44 @@ def capture_stderr
     $stderr = stderr
   end
 
+  def test_list_of_hosts_with_first_host_successful
+    hosts = [
+              ['test.mocked.com', 636],
+              ['test2.mocked.com', 636],
+              ['test3.mocked.com', 636],
+            ]
+    flexmock(TCPSocket).should_receive(:new).ordered.with(*hosts[0]).once.and_return(nil)
+    flexmock(TCPSocket).should_receive(:new).ordered.never
+    Net::LDAP::Connection.new(:hosts => hosts.to_enum(:each))
```

Comment:
That was a late night edit, I think. I was trying to make sure that the
object passed in *only* responded to the `each` method so that the test
enforces the contract and doesn't allow the code to start using other
methods.

For now, I'll probably just chuck the `to_enum` calls unless you would
prefer the enforcement. It shouldn't too hard to make a simple wrapper
class to mask out the other methods though.

On Mon, Sep 28, 2015, 15:12 Jerry Cheung <notifications@github.com> wrote:

> In test/test_ldap_connection.rb
> <https://github.com/ruby-ldap/ruby-net-ldap/pull/223#discussion_r40599218>
> :
>
> > @@ -9,6 +9,44 @@ def capture_stderr
> >      $stderr = stderr
> >    end
> >
> > +  def test_list_of_hosts_with_first_host_successful
> > +    hosts = [
> > +              ['test.mocked.com', 636],
> > +              ['test2.mocked.com', 636],
> > +              ['test3.mocked.com', 636],
> > +            ]
> > +    flexmock(TCPSocket).should_receive(:new).ordered.with(*hosts[0]).once.and_return(nil)
> > +    flexmock(TCPSocket).should_receive(:new).ordered.never
> > +    Net::LDAP::Connection.new(:hosts => hosts.to_enum(:each))
>
> Why to_enum here? hosts is already an iterable object?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ruby-ldap/ruby-net-ldap/pull/223/files#r40599218>.
>